config/config.go

```diff
@@ -44,8 +44,9 @@ type Config struct {
 	Insecure           bool
 	Protector          pnet.Protector
 
-	Relay     bool
-	RelayOpts []circuit.RelayOpt
+	RelayCustom bool
```

Comment:
`RelayCustom` seems like an bookkeeping flag we use only to track if the user has set relay options explicitly. Can we make it private?

Reply:
I would have liked  this to, but the readers are in a different package.

Reply:
👍

Reply:
Maybe a docstring would clarify the role of this flag, also stating it's not intended for explicit use.

Reply:
Maybe we should change `Relay` to an enum (ish) with `Enabled`, `Disabled`, and `Default` variants.

Reply:
I thought about that too, but didn’t propose it because it’d imply breakage.

Reply:
This struct actually has an explicit warning saying that it's not stable. Users should use the *options*, they shouldn't modify the config directly.

Reply:
ok, I got started in making it an enum and it's extremely cumbersome. 
Better to keep the RelayCustom contraption.